src/services/axios.ts

```diff
@@ -0,0 +1,28 @@
+import axios from "axios";
+
+const locationAxios = axios.create({
+  baseURL: "https://ipapi.co/json/",
+});
+
+locationAxios.interceptors.request.use(
+  (req) => {
+    console.log(req);
+    return req;
+  },
+  (err) => {
+    return Promise.reject(err);
+  }
+);
+export default locationAxios;
+
+const articlesAxios = axios.create({
+  baseURL: `https://newsapi.org/v2/`,
+  timeout: 10000,
+  headers: { "api-key": "2fe52557b5b44b6e9abd6666a5079799" },
```

Comment:
extract api-key to file called : ".env" (and place it in the root)
and add '.env' to .gitignore